### ISSUE TITLE
No Entrance Transitions for VictoryArea and VictoryLine

### ISSUE THREAD

**Issue by chrisbolin** [FormidableLabs/victory-native#77]
Entrance transitions don't work for VictoryArea and VictoryLine, but animations do. VictoryBar, etc are not effected (transitions work fine)

**Comment by stief510** [comment]
Is there an update on this? I am trying to get the graph to animate when the component first loads but I cannot

**Comment by chrisbolin** [comment]
@stief510 could you tell more about your setup? is this on iOS or Android? What version of the victory packages are you using?

on iOS, entrance transitions should work as of `victory-native@0.9.0`, `victory-chart@20.0.0`. Here's a quick demo...

![jun-07-2017 22-15-26 - onload](https://user-images.githubusercontent.com/2822048/26901884-2df3a2ba-4bcf-11e7-8211-ce01e5dbfe96.gif)

**Comment by stief510** [comment]
I have only tried on iOS thus far. I am using

```
        <VictoryArea
          data={data}
          domain={{
            x: domainx,
            y: domainy
          }}
          x="date"
          y={(d) => _.toNumber(d.value)}
          interpolation="monotoneX"
          scale={{x: "time", y: "linear"}}
          standalone={false}
          style={styles.lineOne}
          animate={{
            onLoad: {
              duration: 2000
            }
          }}
        />
```

**Comment by chrisbolin** [comment]
@stief510 thanks for the quick response. i see the issue with both VictoryArea and VictoryLine. i'll look into it

**Comment by chrisbolin** [comment]
changed the description of the issue, too. After looking at it, I really don't think we broke entrance transitions with #74; most likely they never worked.

**Comment by chrisbolin** [comment]
I'll leave some notes as I debug.

Found that `props.clipWidth` in native's `clip-path.js` is always undefined. Probably not what we want :)

**Comment by chrisbolin** [comment]
I believe the core of the problem is this:

The clipping seems to be completely static, based on the first values given to `ClipPath`. Therefore no animation works. The first value is the full width, the fix I made in #74.

E.g. the width of the clip container should be 0, 1, 2, ... 275. But _effectively_ it is always the first number. I saw this behavior earlier with #73, but here it's even more troublesome, becomes we want a dynamic width.

I've traced this all the way through to `react-native-svg`, and its `ClipPath` renders a `Rect` with dynamic width. I'm not sure where the bug is coming from, but I'll keep digging.

**Comment by chrisbolin** [comment]
Alright, I can reproduce this with `react-native-svg` alone, so that narrows it a lot. I'll look through their issues / fixes and see if I can find anything.

**Comment by chrisbolin** [comment]
`react-native-svg` bug reproduction...

```jsx
import React, { Component } from "react";
import { ScrollView } from "react-native";
import Svg, { Defs, Rect, ClipPath, Circle } from "react-native-svg";

export default class Demo extends Component {
  constructor(props) {
    super(props);
    this.state = {
      clipWidth: 50
    };
  }
  componentDidMount() {
    setInterval(
      () => this.setState({ clipWidth: this.state.clipWidth + 10 }),
      1000
    );
  }
  render() {
    return (
      <ScrollView>
        <Svg height="100" width="100">
            <Defs>
                <ClipPath id="clip">
                    <Rect x="0" y="0" width={this.state.clipWidth} height="100" />
                </ClipPath>
            </Defs>
            <Circle cx="50" cy="50" r="50" clipPath="url(#clip)" />
        </Svg>
      </ScrollView>
    );
  }
}
```

**Comment by stief510** [comment]
Thanks!

**Comment by chrisbolin** [comment]
oh nice! it looks like this is solved in `react-native-svg@5.2.0`. a side effect is that we need to upgrade to `react-native@0.45.0` and `react@16.0.0-alpha.12`. I'll make sure that fixes the issue in `victory-native`, and then release a patch

**Comment by chrisbolin** [comment]
fixed in PR https://github.com/FormidableLabs/victory-native/pull/99. i'll update this thread when a new version is published.

**Comment by chrisbolin** [comment]
released as `victory-native@0.10.0`. cc @stief510 
note that the peer deps changed to `react-native-svg@5.2.0` `react-native@0.45.0` `react@16.0.0-alpha.12`
